ahl_targets/analysis/2024_update/output_comparison.py

```diff
@@ -0,0 +1,304 @@
+    )
+
+    # Log the results
+    logging.info("New model swa by store")
+    logging.info(new_swa_by_store)
+    logging.info("Old model swa by store")
+    logging.info(old_swa_by_store)
+
+    # Insights:
+    #    - The reductions are all of a similar magnitude by store, although in general the new model leads to slightly larger reductions. Mean reduction: 0.695 vs 0.664. Median reduction: 0.701 vs 0.666.
+    #    - The new model has an increased baseline NPM for all stores. The most notable are Iceland, Co-op and Waitrose with baseline increases of 0.4. We would expect this to be as a result of relatively higher sales of re-included products.
+
+    ## Verifying the increase in baseline NPM: Iceland, Co-op and Waitrose
+    # Get the old and new model data
+    logging.info("Loading old and new model data")
+
+    # Old
+    store_data = get_data.model_data()
+    prod_table = get_data.product_metadata()
+
+    store_weight_npm = su.weighted_npm(store_data)
+    store_weight_npm["prod_weight_g"] = store_weight_npm.pipe(su.prod_weight_g)
+
+    old_model_data = store_weight_npm.copy()
+
+    # New
+    new_model_data = g2.get_agg_data_2024()
+
+    # Filter to the stores of interest
+    store_filter = ["Total Iceland", "The Co-Operative", "Total Waitrose"]
+    old_model_data = old_model_data[
+        old_model_data["store_cat"].isin(store_filter)
+    ].reset_index(drop=True)
+    new_model_data = new_model_data[
+        new_model_data["store_cat"].isin(store_filter)
+    ].reset_index(drop=True)
+
+    # Examine the new products
+    new_products = new_model_data[
+        ~new_model_data.product_code.isin(old_model_data.product_code.unique())
+    ].reset_index(drop=True)
+
+    # Group by store and calculate swa of NPM for new products and compare to the swa across all products
+    new_products_swa = (
+        new_products.groupby("store_cat")
+        .apply(lambda x: np.average(x["npm_score"], weights=x["total_kg"]))
+        .reset_index()
+        .rename(columns={0: "new_products_swa"})
+    )
+    all_products_swa = (
+        new_model_data.groupby("store_cat")
+        .apply(lambda x: np.average(x["npm_score"], weights=x["total_kg"]))
+        .reset_index()
+        .rename(columns={0: "all_products_swa"})
+    )
+    new_products_swa = new_products_swa.merge(
+        all_products_swa, on="store_cat", how="left"
+    )
+
+    # Log the results
+    logging.info("New products swa by store")
+    logging.info(
+        new_products_swa
+    )  # SWA of new products is *much* higher. We expect these to be oils, icecreams etc.
+
+    ## Further investigation:
+    # What exactly are these products?
+
+    # Load the silver product data to merge on
+    ## Flag: Due to python version issues I can't seem to use ahl core data - accessing the file directly from the s3 bucket instead
+    prod = download_obj(
+        BUCKET_NAME,
+        "kantar/silver/inhome_products_silver_full.parquet",
```

Comment:
I don't think it will change anything but would switch to using the original files in case:

```suggestion
        "kantar_orig/silver/inhome_products_silver.parquet",
```